model/src/main/proto/feedback_reporting.proto

```diff
@@ -0,0 +1,136 @@
+  int64 report_date_ms = 1;
+  FeedbackReportingForm feedback_form = 2;
+  FeedbackReportingDeviceInfo device_info = 3;
+  FeedbackReportingBuildInfo build_info = 4;
+  FeedbackReportingAppInfo app_info = 5;
+}
+
+// Information supplied by the user in the feedback report.
+message FeedbackReportingForm {
+  oneof report_type {
+    Suggestion suggestion = 1;
+    Issue issue = 2;
+    Crash crash = 3;
+  }
+}
+
+// A suggestion feedback report.
+message Suggestion {
+  enum SuggestionCategory {
+    CATEGORY_UNSPECIFIED = 0;
+    FEATURE_SUGGESTION = 1;
+    LANGUAGE_SUGGESTION = 2;
+    OTHER_SUGGESTION = 3;
+  }
+  SuggestionCategory suggestion_category = 1;
+  // The text input provided by the user.
+  string user_submitted_suggestion = 2;
+}
+
+// An issue feedback report.
+message Issue {
+  enum IssueCategory {
+    CATEGORY_UNSPECIFIED = 0;
+    LESSONS_QUESTIONS_ISSUE = 1;
+    LANGUAGE_ISSUE = 2;
+    LANGUAGE_AUDIO_ISSUE = 3;
+    LANGUAGE_TEXT_ISSUE = 4;
+    TOPICS_ISSUE = 5;
+    PROFILE_ISSUE = 6;
+    OTHER_ISSUE = 7;
+  }
+  IssueCategory issue_category = 1;
+  // List of checkbox items selected by user in the feedback report.
+  repeated string issues_list = 2;
+  // The user's text input if they select "other" for any option
+  string other_user_input = 3;
+}
+
+// A crash report.
+message Crash {
+  enum CrashLocation {
+    LOCATION_UNSPECIFIED = 0;
+    LESSON_PLAYER = 1;
+    PRACTICE_QUESTIONS = 2;
+    TOPICS_LIST = 3;
+    OPTIONS_ACTIVITY = 4;
+    PROFILE_ACTIVITY = 5;
+    OTHER = 6;
+  }
+  CrashLocation crash_location = 1;
+  // The user's text input for explaining what happened.
+  string crash_explanation = 2;
+}
+
+// Information collected about the user's device.
+message FeedbackReportingDeviceInfo {
+  string package_name = 1;
+  string package_code = 2;
+  string country_locale = 3;
+  string language_locale = 4;
+}
+
+// Information collected about the device's build.
+message FeedbackReportingBuildInfo {
+  enum NetworkType {
+    NETWORK_TYPE_UNSPECIFIED = 0;
+    WIFI = 1;
+    CELLULAR = 2;
+  }
+  string device_model = 1;
+  string sdk_version = 2;
+  string device_brand = 3;
+  string build_fingerprint = 4;
+  string phone_type = 5;
+  NetworkType network_type = 6;
+  string device_language = 7;
+}
+
+// Information collected about the app's build.
+message FeedbackReportingAppInfo {
+  enum EntryPoint {
+    ENTRY_POINT_UNSPECIFIED = 0;
+    MAIN_MENU = 2;
+    EXPLORATION = 3;
+    CRASH = 4;
+  }
+  enum TopicProgressStatus {
+    // The completion status is unknown.
+    STATUS_UNSPECIFIED = 0;
+    // No stories have been started in this topic.
+    NOT_STARTED = 1;
+    // The topic has been started but not completed by the player.
+    STARTED_NOT_COMPLETED = 2;
+    // The topic has been completed by the player.
+    COMPLETED = 3;
+  }
+
+  EntryPoint entry_point = 1;
+  // Maps topic ID to the completion state of topic, based on state of stories in topics
```

Comment:
Suggest also mentioning a bit about why this is here (e.g. to capture the learner's saved lesson state to have additional context).

That being said, maybe 'context' is a bit clearer than 'info' for the protos here. WDYT?

Reply:
Ack, I like the using context because that is the goal of collecting all this information -- to gain understanding of the context in which the report is filed in. Will change the names